### ISSUE TITLE
Reordering on more than 1 variable

### ISSUE THREAD

**Issue by hochoy** [STAT545-UBC/Discussion#228]
Hi,

I am trying to re-order my factor levels using more than 1 variable. E.g. reorder continent first by country and then by gdp_percapita. I know arrange() can take more than 1 variable but I have't been able to do the same with reorder().

E.g. arrange(gapminder, country, gdp_percapita)   # sorts by country first then gdp_percapita
E.g. gapminder2 <- gapminder %>% mutate(continent = reorder(continent, gdpPercap))     WORKS!
E.g. gapminder3 <- gapminder %>% mutate(continent = reorder(continent, country, gdpPercap)) DOESNT WORK!

Is there a workaround so that I can re-order continent's factor levels based on more than 1 variable? Thanks!

**Comment by jennybc** [comment]
At first I thought "great question"! Then I started to work through it concretely and realized we need to get more specific.

What does it even mean to reorder the continent factor by country?

But assuming we did that, what would it mean to further reorder it by gdpPercap? You're assuming that there could be ties based on the previous ordering and you want to use the second variable to break them?

Help me understand what you're asking.

I guess I can imagine needing access to more than one companion variable in order to compute the summary statistics that will drive factor reordering. And `reorder()` does NOT support that. But I suspect that's not what you meant.
